src/main/java/de/pierreschwang/spigotlib/inventory/SimplePaginatedInventory.java

```diff
@@ -1,41 +1,116 @@
+import de.pierreschwang.spigotlib.inventory.exceptions.PaginatedInventoryException;
 import de.pierreschwang.spigotlib.item.ItemFactory;
+import org.bukkit.Material;
 import org.bukkit.entity.Player;
 import org.bukkit.event.inventory.InventoryClickEvent;
 import org.bukkit.inventory.ItemStack;
 
-import java.util.LinkedHashMap;
-import java.util.OptionalInt;
+import java.util.*;
 import java.util.function.Consumer;
-import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 public class SimplePaginatedInventory extends SimpleInventory {
+    private static final ItemStack ARROW_LEFT_ITEM = ItemFactory.skull()
+            .texture("eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYmQ2OWUwNmU1ZGFkZmQ4NGU1ZjNkMWMyMTA2M2YyNTUzYjJmYTk0NWVlMWQ0ZDcxNTJmZGM1NDI1YmMxMmE5In19fQ==")
+            .name("§6<-")
+            .apply();
+    private static final ItemStack ARROW_RIGHT_ITEM = ItemFactory.skull()
+            .texture("eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvMTliZjMyOTJlMTI2YTEwNWI1NGViYTcxM2FhMWIxNTJkNTQxYTFkODkzODgyOWM1NjM2NGQxNzhlZDIyYmYifX19")
+            .name("§6->")
+            .apply();
+    private int pageSwitcherBackSlot = -1;
+    private int pageSwitcherForwardSlot = -1;
     private final LinkedHashMap<Integer, ItemStack> inventoryContents;
+    private final List<Integer> dynamicSlots;
     private int currentPage;
 
-    SimplePaginatedInventory(int size, String title) {
+    SimplePaginatedInventory(int size, String title, int... dynamicSlots) {
         super(size, title);
-        inventoryContents = new LinkedHashMap<>();
-        currentPage = 1;
-    }
-
-    SimplePaginatedInventory(int size) {
-        this(size, "Inventory");
+        this.currentPage = 1;
+        this.inventoryContents = new LinkedHashMap<>();
+        this.dynamicSlots = Arrays.stream(dynamicSlots).boxed().collect(Collectors.toList());
+        Collections.sort(this.dynamicSlots);
+        if(this.dynamicSlots.size() <= 0) {
+            return;
+        }
+        if(this.dynamicSlots.get(0) < 0) {
+            throw new PaginatedInventoryException("Dynamic slot out of range: value for a dynamic slot can not be less than 0");
+        }
+        if(this.dynamicSlots.get(this.dynamicSlots.size() - 1) >= getInventory().getSize()) {
+            throw new PaginatedInventoryException("Dynamic slot out of range: value for dynamic slot can not be greater than the size of the inventory");
+        }
     }
 
     public LinkedHashMap<Integer, ItemStack> getInventoryContents() {
         return inventoryContents;
     }
 
+    public List<Integer> getDynamicSlots() {
+        return dynamicSlots;
+    }
+
+    public void addDynamicSlots(int... slots) {
+        getDynamicSlots().addAll(Arrays.stream(slots).boxed().collect(Collectors.toList()));
```

Comment:
I suggest to use Arrays#asList instead of creating a new IntStream and collect it.